src/anyio/_core/_compat.py

```diff
@@ -13,7 +13,7 @@
 
 T = TypeVar('T')
 AnyDeprecatedAwaitable = Union['DeprecatedAwaitable', 'DeprecatedAwaitableFloat',
-                               'DeprecatedAwaitableList', TaskInfo]
+                               'DeprecatedAwaitableList[T]', TaskInfo]
```

Comment:
`AnyDeprecatedAwaitable` with `DeprecatedAwaitableList[T]` will become generic type alias. Are you sure this is what is needed?

See example:

```
In [3]: T = TypeVar('T')

In [4]: X = Union[int, List[T]]

In [5]: X
Out[5]: typing.Union[int, typing.List[~T]]

In [6]: X[str]
Out[6]: typing.Union[int, typing.List[str]]
```

Reply:
@erictraut, why pyright doesn't complain about usages of `AnyDeprecatedAwaitable` without specifying explicit type arguments (in `maybe_async` for example)?

Reply:
I would expect pyright to complain about that. If that's not what you're seeing, please file a bug with a repro description, and I'll investigate further. Thanks!

Reply:
> AnyDeprecatedAwaitable with DeprecatedAwaitableList[T] will become generic type alias. Are you sure this is what is needed?

The third overload was already using a type variable. The fix just added that to the union so it matches with the overload.

Reply:
I meant the third item in the return type union: `List[T]`.

Reply:
But now you must specify type argument for all usages of `AnyDeprecatedAwaitable`, because now it is genetic too. I think `DeprecatedAwaitableList[Any]` is what you need.

Reply:
Sorry, I haven't seen the latest changes...